src/SALib/analyze/morris.py

```diff
@@ -240,47 +195,86 @@ def compute_elementary_effects(model_inputs, model_outputs, trajectory_size,
     ---------
     ee : np.array
         Elementary Effects for each parameter
-    '''
+    """
     num_vars = model_inputs.shape[1]
     num_rows = model_inputs.shape[0]
     num_trajectories = int(num_rows / trajectory_size)
 
-    ee = np.zeros((num_trajectories, num_vars), dtype=np.float)
-
     ip_vec = model_inputs.reshape(num_trajectories, trajectory_size, num_vars)
     ip_cha = np.subtract(ip_vec[:, 1:, :], ip_vec[:, 0:-1, :])
     up = (ip_cha > 0)
     lo = (ip_cha < 0)
 
     op_vec = model_outputs.reshape(num_trajectories, trajectory_size)
 
-    result_up = get_increased_values(op_vec, up, lo)
-    result_lo = get_decreased_values(op_vec, up, lo)
+    result_up = _get_increased_values(op_vec, up, lo)
+    result_lo = _get_decreased_values(op_vec, up, lo)
 
     ee = np.subtract(result_up, result_lo)
     np.divide(ee, delta, out=ee)
 
     return ee
 
 
-def compute_mu_star_confidence(ee, num_trajectories, num_resamples,
-                               conf_level):
-    '''
+def _compute_mu_star_confidence(elementary_effects: np.ndarray, num_vars: int,
+                                num_resamples: int, conf_level: float):
+    """
```

Comment:
While you're here, it would be good to add some docstring here in the numpydoc format and description of the arguments for completeness!

```python
"""Numpydoc has a one line active description of the function

Then there is one line space, followed by a paragraph of free text 
with a more extensive description

Arguments
-----------
elementary_effects : numpy.ndarray
    A description of the argument
"""
```

Reply:
Hi, @willu47. It's my pleasure to help.

I included the docstrings, and took the chance to add some extra functions too. I also made adaptations to the tests, so now they cover all functions. Some more trivial functions are indirectly tested.

I included a function called `_calc_results_difference`. At the moment this function is trivial, it only has one line of code (`results_difference = np.subtract(result_up, result_lo)`). Nevertheless, I'll open an issue today to include a new functionality, and this abstraction would be necessary, so I decided to do it in this PR.